src/dso_api/dynamic_api/serializers.py

```diff
@@ -285,10 +299,109 @@ def _profile_based_authorization_and_mutation(self, data):
+            ]
+        )
+        return output_fields
+
+    def to_representation(self, validated_data):
+        """Restrict the output from the DynamicSerializer"""
+        data = super().to_representation(validated_data)
+        output_fields = self.get_output_fields()
+        output_data = OrderedDict(
+            [
+                (field_name, field)
+                for field_name, field in data.items()
+                if field_name in output_fields or field_name == "_embedded"
+            ]
+        )
+        return output_data
+
+
+class DynamicLinksSerializer(DynamicSerializer):
+    """The serializer for the _links field. Following DSO/HAL guidelines, it contains "self",
+    "schema", and all relational fields that have not been expanded. This depends on the
+    _expand and _expandScope URL parameters. When expanded, they move to "_embedded".
+    In case of a list-view, keep the relational fields. The resources in "_embedded" are
+    grouped together for deduplication purposes and this will allow the user to know which
+    embed belongs to which object.
+    """
+
+    serializer_related_field = HALTemporalHyperlinkedRelatedField
+
+    def get_output_fields(self):
+        """The list of fields to include in the serialization"""
+        fields = self.fields
+        embedded_fields = self.get_fields_to_expand()
+        link_fields = ["self", "schema"]
+        relation_fields = [
+            field_name
+            for field_name, field in fields.items()
+            if isinstance(
+                field,
+                (HyperlinkedRelatedField, ManyRelatedField, LooseRelationUrlField),
+            )
+        ]
+        # add the relation_fields to _links if there is not going to be an _embedded
+        # or if we are part of a list-view
+        if embedded_fields is False or (
+            "view" in self.context
+            and hasattr(self.context["view"], "detail")
+            and self.context["view"].detail is False
+        ):
+            link_fields += relation_fields
+        elif isinstance(embedded_fields, list):
+            link_fields += [
+                field_name
+                for field_name in relation_fields
+                if field_name not in embedded_fields
+            ]
+        output_fields = OrderedDict(
+            [
+                (field_name, field)
+                for field_name, field in fields.items()
+                if field_name in link_fields
+            ]
+        )
+        return output_fields
+
+    def to_representation(self, validated_data):
+        """Restrict the output from the DynamicSerializer"""
+        data = super().to_representation(validated_data)
+        output_fields = self.get_output_fields()
+        output_data = OrderedDict(
+            [
+                (field_name, field)
+                for field_name, field in data.items()
+                if field_name in output_fields
+            ]
+        )
+        return output_data
```

Comment:
Handiger is `self.fields` aan te passen, zodat er niet onnodig data uitgelezen wordt. Dan hoef je de velden niet achteraf eruit te slopen.

Reply:
Dit was handig voor temporele data, vermijdt elders weer extra logica. Vermoedelijk is er geen onnodige data uitlezing vanwege het Django lazy loading model, maar hier zal ik nog naar kijken; goed punt.